lib/store_model/ext/parent_assignment.rb

```diff
@@ -7,9 +7,9 @@ module ParentAssignment
 
     def assign_parent_to_store_model_relation(attribute)
       assign_parent_to_singular_store_model(attribute)
-      return unless attribute.is_a?(Array)
+      return if !attribute.is_a?(Array) && !attribute.is_a?(Hash)
 
-      attribute.each(&method(:assign_parent_to_singular_store_model))
+      (attribute.try(:values) || attribute).each(&method(:assign_parent_to_singular_store_model))
```

Comment:
I'm using ActiveSupport's #try here because it's included in ActiveRecord. LMK if you don't want to use that transitive dependency and I can just use `respond_to?(:values)` instead